src/Tests/dotnet-workload-install.Tests/WorkloadGarbageCollectionTests.cs

```diff
@@ -0,0 +1,393 @@
+            CreateMockManifest("testmanifest", "2.0.0", "6.0.300", sourceManifestName: @"Sample2_v2.json");
+            CreateMockManifest("testmanifest", "3.0.0", "6.0.300", sourceManifestName: @"Sample2_v3.json");
+
+            CreateManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300");
+            CreateManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300");
+            CreateManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300");
+
+            var (installer, getResolver) = GetTestInstaller("6.0.300");
+
+            var packsToKeep = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk),
+                CreatePackInfo("Xamarin.Android.Framework", "8.6.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.6.0.0", WorkloadPackKind.Library)
+            };
+
+            var packsToCollect = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Framework", "8.4.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.4.7.4", WorkloadPackKind.Library)
+            };
+
+            foreach (var pack in packsToKeep.Concat(packsToCollect))
+            {
+                CreateInstalledPack(pack, "6.0.300");
+            }
+
+            // Write workload install record for 6.0.300
+            var workloadsRecordPath = Path.Combine(_dotnetRoot, "metadata", "workloads", "6.0.300", "InstalledWorkloads");
+            Directory.CreateDirectory(workloadsRecordPath);
+            File.Create(Path.Combine(workloadsRecordPath, "xamarin-android-build"));
+
+            installer.GarbageCollect(getResolver);
+
+            foreach (var pack in packsToCollect)
+            {
+                PackShouldExist(pack, false);
+                PackRecord(pack, "6.0.300").Should().NotExist();
+            }
+            foreach (var pack in packsToKeep)
+            {
+                PackShouldExist(pack, true);
+                PackRecord(pack, "6.0.300").Should().Exist();
+            }
+
+            ManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300").Should().NotExist();
+            ManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300").Should().NotExist();
+            ManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300").Should().Exist();
+
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.100", "testmanifest", "1.0.0", "WorkloadManifest.json")).Should().NotExist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "2.0.0", "WorkloadManifest.json")).Should().NotExist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "3.0.0", "WorkloadManifest.json")).Should().Exist();
+        }
+
+        [Fact]
+        public void GarbageCollectManifestsWithInstallState()
```

Comment:
This test is rather very similar to `GarbageCollectManifests`, it's very likely better to write it as a Theory with a boolean for `installState`.

Reply:
That's what I started out trying to do when writing this test, but I realized that the `packsToKeep` and `packsToCollect` would have to differ, as well as the assertions for which manifests and manifest records should exist.  With so many conditionals in the Theory, it seemed like it might be better to write it as a separate test, so I ended up going with that.

Reply:
That's fair!